iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -84,132 +74,341 @@ public double getA() {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y1 = new Complex(g1, b1); // y1=g1+jb1
-        Complex y2 = new Complex(g2, b2); // y2=g2+jb2
-        Complex s2 = computeS2(y1, y2, z, rho);
-        return -s2.getImaginary();
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
-    public double otherSideQ(TieLine.HalfLine hl) {
-        return otherSideQ(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), hl.getG2(), hl.getB2(), 1.0);
+    public double otherSideQ(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSideQ(adm);
     }
 
+    // Has been modified
+    // Before: otherSideQ(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, dl.getG() / 2.0, dl.getB() / 2.0, 1.0, 0.0);
     public double otherSideQ(DanglingLine dl) {
-        return otherSideQ(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, dl.getG() / 2.0, dl.getB() / 2.0, 1);
+        return otherSideQ(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+    }
+
+    public double otherSideQ(DanglingLine dl, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSideQ(dl.getR(), dl.getX(), dl.getG() * 0.5, dl.getB() * 0.5, dl.getG() * 0.5, dl.getB() * 0.5, 1.0, 0.0);
+        } else {
+            return otherSideQ(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+        }
+    }
+
+    public double otherSideQ(TieLine.HalfLine hl) {
+        return otherSideQ(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), hl.getG2(), hl.getB2(), 1.0, 0.0);
     }
 
+    /**
+     * @deprecated Not used anymore. Use
+     * {@link SV#otherSideU(double, double, double, double, double, double, double, double)}
+     */
+    @Deprecated
     public double otherSideU(double r, double x, double g1, double b1, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y1 = new Complex(g1, b1); // y1=g1+jb1
-        Complex u2 = computeU2(y1, new Complex(0, 0), z, rho);
-        return u2.abs();
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
-    public double otherSideU(TieLine.HalfLine hl) {
-        return otherSideU(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), 1.0);
+    public double otherSideU(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSideU(adm);
     }
 
+    // Has been modified
+    // Before: otherSideU(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, 0.0, 0.0, 1.0, 0.0);
```

Comment:
no need to comment previous implementation

Reply:
Done